analyses/independent-samples-release/util/independent-rna-samples.R

```diff
@@ -0,0 +1,118 @@
+                  match_id %in% independent_dna$match_id)
+  
+  # has rna samples which match the independent samples provided
+  sample_df <- matched_rna
+
+  
+  # Here we are adding only initial only-RNA-Seq samples
+  # since this will always to part of independent_dna_plus_only_rna
+  # regardless tumor_description_rna_only is "primary" OR "primary_plus"
+  #
+  if( match_type == "independent_dna_plus_only_rna" ) {
+    # find sample set where we initial only-RNA-Seq samples
+    only_rna_intial <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    tumor_descriptor %in% primary_descs,
+                    # find and remove participants which have 
+                    # matching dna samples in independent_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID)
+    
+    # has rna samples which match the independent samples provided plus rna only sample which are primary tumors
+    sample_df <- bind_rows(sample_df,only_rna_intial)
+  }
+  
+  
+  # Here we are adding only-RNA-Seq samples which are not initial
+  # if tumor_description_rna_only == "primary_plus"
+  #
+  if(match_type == "independent_dna_plus_only_rna" & tumor_description_rna_only == "primary_plus"){
+    # find sample set where we only find rna samples
+    only_rna_plus <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    # find and remove participants which have 
+                    # matching dna samples in independend_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                    # and participant not in only_rna_initial sample set
+                    !Kids_First_Participant_ID %in% only_rna_intial$Kids_First_Participant_ID
+                    )
+    # has rna samples which match the independent samples provided plus rna only sample which are primary tumors plus rna samples where no primary primaries exists
+    sample_df <- bind_rows(sample_df,only_rna_plus)
+  } 
+  
+  # get the samples from the earliest timepoints for each Participant
+  # age_at_diagnosis_days is no longer relevant,
+  # as it is the same for all samples from an participant, but
+  # leaving this in for future use in case we get specimen order data 
+  early_samples <- sample_df %>%
+    dplyr::group_by(Kids_First_Participant_ID) %>%
+    dplyr::summarize(age_at_diagnosis_days = min(as.numeric(age_at_diagnosis_days))) %>%
+    dplyr::left_join(sample_df, by = c("Kids_First_Participant_ID",
+                                       "age_at_diagnosis_days"))
+  
+  # Choose randomly among specimens from the same participant
+  early_ind <- early_samples %>%
+    dplyr::group_by(Kids_First_Participant_ID, match_id) %>%
+    dplyr::summarize(Kids_First_Biospecimen_ID = sample(Kids_First_Biospecimen_ID, 1)) 
```

Comment:
```suggestion
    dplyr::summarize(Kids_First_Biospecimen_ID = sample(Kids_First_Biospecimen_ID, 1)) %>%
    dplyr::ungroup() %>%
    dplyr::select(Kids_First_Participant_ID, Kids_First_Biospecimen_ID)
```

Similar to the comment on the returned table in independent DNA function.